tsdb/exemplar.go

```diff
@@ -57,16 +62,28 @@ func NewCircularExemplarStorage(len int, reg prometheus.Registerer) (ExemplarSto
 	c := &CircularExemplarStorage{
 		exemplars: make([]*circularBufferEntry, len),
 		index:     make(map[string]*indexEntry),
-		outOfOrderExemplars: prometheus.NewCounter(prometheus.CounterOpts{
+		exemplarsAppended: promauto.With(reg).NewCounter(prometheus.CounterOpts{
```

Comment:
Can we continue with `MustRegister` for now? We're not using promauto anywhere in tsdb at then moment and we were trying to remove usage of it in remote write IIRC.

Reply:
Why remove this? Just curious.

Reply:
Sorry, I don't remember all of the context for why. I think it's in part because we want to pass structs for the metrics in in more places where we expect other devs to use our code like a library, in which case both promauto and `MustRegister` are problematic if the code is being reused in something like a multitenant storage system :)

Reply:
I think you needed to unregister at some point or something (: